lib/ansible/utils/vars.py

```diff
@@ -128,6 +128,8 @@ def load_extra_vars(loader):
         if extra_vars_opt.startswith(u"@"):
             # Argument is a YAML file (JSON is a subset of YAML)
             data = loader.load_from_file(extra_vars_opt[1:])
+        elif extra_vars_opt and extra_vars_opt[0] in u'/.':
+            raise AnsibleOptionsError("Please prepend extra_vars filename with '@'")
```

Comment:
you should check for `if extra_vars_opt` on top and `continue` otherwise, if it would fail here, we could have a exception on the previous startswith (if `None`), also that lets us change it to a array ref on all counts